roofit/roofitcore/src/RooBinSamplingPdf.cxx

```diff
@@ -199,6 +199,56 @@ RooSpan<const double> RooBinSamplingPdf::binBoundaries() const {
 }
 
 
+////////////////////////////////////////////////////////////////////////////////
+/// Return a list of all bin boundaries, so the PDF is plotted correctly.
+/// \param[in] obs Observable to generate the boundaries for.
+/// \param[in] xlo Beginning of range to create list of boundaries for.
+/// \param[in] xhi End of range to create to create list of boundaries for.
+/// \return Pointer to a list to be deleted by caller.
+std::list<double>* RooBinSamplingPdf::binBoundaries(RooAbsRealLValue& obs, Double_t xlo, Double_t xhi) const {
+  if (obs.namePtr() != _observable->namePtr()) {
+    coutE(Plotting) << "RooBinSamplingPdf::binBoundaries(" << GetName() << "): observable '" << obs.GetName()
+        << "' is not the observable of this PDF ('" << _observable->GetName() << "')." << std::endl;
+    return nullptr;
+  }
+
+  auto list = new std::list<double>;
+  for (double val : binBoundaries()) {
+    if (xlo <= val && val < xhi)
+      list->push_back(val);
+  }
+
+  return list;
```

Comment:
And who `delete`s this thing?

Reply:
Don't know how the interface is used in other places, but maybe it's better to use smart ptrs?

Reply:
I wish ...

As you can see, it's `override`, so I cannot change it ... The caller has to delete it.